pkg/admissioncontroller/webhook/admission/resourcesize/handler.go

```diff
@@ -31,94 +34,78 @@ import (
 	apisconfig "github.com/gardener/gardener/pkg/admissioncontroller/apis/config"
 	confighelper "github.com/gardener/gardener/pkg/admissioncontroller/apis/config/helper"
 	"github.com/gardener/gardener/pkg/admissioncontroller/metrics"
-	acadmission "github.com/gardener/gardener/pkg/admissioncontroller/webhook/admission"
-	"github.com/gardener/gardener/pkg/logger"
 )
 
-const (
-	// HandlerName is the name of this admission webhook handler.
-	HandlerName = "resource_size_validator"
-	// WebhookPath is the HTTP handler path for this admission webhook handler.
-	WebhookPath = "/webhooks/validate-resource-size"
+// metricReasonSizeExceeded is a metric reason value for a reason when an object size was exceeded.
+const metricReasonSizeExceeded = "Size Exceeded"
 
-	// metricReasonSizeExceeded is a metric reason value for a reason when an object size was exceeded.
-	metricReasonSizeExceeded = "Size Exceeded"
-)
-
-// New creates a new webhook handler validating that the resource size of a request doesn't exceed the configured
-// limits.
-func New(logger logr.Logger, config *apisconfig.ResourceAdmissionConfiguration) *handler {
-	return &handler{logger: logger, config: config}
-}
-
-type handler struct {
-	logger  logr.Logger
-	config  *apisconfig.ResourceAdmissionConfiguration
-	decoder *admission.Decoder
-}
-
-var _ admission.Handler = &handler{}
-
-func (h *handler) InjectDecoder(d *admission.Decoder) error {
-	h.decoder = d
-	return nil
+// Handler checks the resource sizes.
+type Handler struct {
+	Logger logr.Logger
+	Config *apisconfig.ResourceAdmissionConfiguration
 }
 
-func (h *handler) Handle(_ context.Context, request admission.Request) admission.Response {
-	requestLogger := logger.NewIDLogger(h.logger.WithValues(
-		"user", request.UserInfo.Username,
-		"resource", request.Resource, "name", request.Name,
-	))
-	if request.Namespace != "" {
-		requestLogger = requestLogger.WithValues("namespace", request.Namespace)
+// Handle checks the resource sizes.
+func (h *Handler) Handle(_ context.Context, req admission.Request) admission.Response {
+	var err error
+	switch req.Operation {
+	case admissionv1.Create:
+		err = h.handle(req)
+	case admissionv1.Update:
+		err = h.handle(req)
+	default:
```

Comment:
The deletion operation is not handled, so it seems to be reasonable to return an error instead of silently accepting it.

Reply:
Maybe I am mistaken here,  but this webhook runs for every object, and if the object size is greater than the max limit it discards that object. So when an object is marked for deletion this webhook will be invoked and webhook will block the deletion of the object cause it returns an error instead of nil on the deletion request.

Reply:
The webhook only handles `CREATE` and `UPDATE` operations, so there should be no need to handle the `DELETE` operation: https://github.com/gardener/gardener/blob/33dc3fc1bbfc0f07b471e39f384431f0989a09eb/charts/gardener/controlplane/charts/application/templates/validatingwebhook-admission-controller.yaml#L73-L75